src/Castle.Core/DynamicProxy/Generators/ClassProxyGenerator.cs

```diff
@@ -113,7 +108,6 @@ protected virtual Type GenerateType(string name, Type[] interfaces, INamingScope
 			// target is not an interface so we do nothing
 
 			var targetInterfaces = targetType.GetAllInterfaces();
-			var additionalInterfaces = TypeUtil.GetAllInterfaces(interfaces);
```

Comment:
This call is superfluous since the same transform already happens in the `BaseProxyGenerator` ctor.